examples/dev/triangle_edge.py

```diff
@@ -0,0 +1,259 @@
+
+    Parameters:
+    n (int): The number of sides (vertices).
+    radius (float): The radius of the circumscribed circle. Default is 1.
+
+    Returns:
+    np.ndarray: An array of shape (n, 2) containing the vertices coordinates.
+    """
+    angles = np.linspace(0, 2 * np.pi, n, endpoint=False)
+    return np.column_stack((radius * np.cos(angles), radius * np.sin(angles)))
+
+
+# @numba.njit
+def generate_order_vectors(path_, closed) -> np.ndarray:
+    """Generate the vectors tangent to the path."""
+    n = path_[1:] - path_[:-1]
+    norm = np.linalg.norm(n, axis=1, keepdims=True)
+    normals = n / norm
+    vec = np.empty((path_.shape[0], 2, 2))
+    vec[:, 0] = path_
+    vec[:-1, 1] = normals
+    if closed:
+        vec[-1, 1] = (path_[0] - path_[-1]) / np.linalg.norm(path_[-1] - path_[0])
+    else:
+        vec[-1, 1] = -vec[-2, 1]
+    return vec
+
+# @numba.njit
+def generate_miter_helper_vectors(normal_vector: np.ndarray) -> np.ndarray:
+    """Generate the miter helper vectors.
+
+    for each point on the path, the miter helper vectors are pairs of vectors
+    First vector is the normal vector scaled by 1/2
+    Second vector is the normal vector of previous point scaled by -1/2
+    """
+    vec2 = normal_vector.copy()
+    vec2[:, 1] *= 0.5
+    vec3 = vec2.copy()
+    vec3[:, 1] *= -1
+    vec3[:, 1] = np.roll(vec3[:, 1], 1, axis=0)
+    vec3[:, 0] += vec2[:, 1]
+    return np.concatenate([vec2, vec3], axis=0)
+
+@numba.njit
+def generate_orthogonal_vectors(normal_vector: np.ndarray) -> np.ndarray:
+    """Generate the orthogonal vectors to the normal vectors.
+
+    The orthogonal vector starts at the middle of the normal vector and is orthogonal to it.
+    It has length equal to half of the normal vector.
+    """
+    vec2 = normal_vector.copy()
+    vec2[:, 1] *= 0.5
+    vec5 = vec2.copy()
+    vec5[:, 0] += vec2[:, 1]
+    vec5[:, 1, 0] = -vec2[:, 1, 1]
+    vec5[:, 1, 1] = vec2[:, 1, 0]
+    return vec5
+
+@numba.njit
+def generate_mitter_vectors(mesh) -> np.ndarray:
+    """For each point on path, generate the vectors pointing to adjusted points. Generate the mitter vectors for each point on the path.
+    """
+    vec_points = np.empty((mesh[0].shape[0], 2, 2))
+    vec_points[:, 0] = mesh[0]
+    vec_points[:, 1] = mesh[1]
+    return vec_points
+
+@numba.njit
+def generate_edge_triangle_borders(centers, offsets, triangles) -> np.ndarray:
+    """For each triangle in mesh generate 3 vectors that represent the borders of the triangle.
+    """
+    res = np.empty((len(triangles)*3, 2, 2), dtype=centers.dtype)
+    for i, triangle in enumerate(triangles):
+        a, b, c = triangle
+        a1 = centers[a] + offsets[a]
+        b1 = centers[b] + offsets[b]
+        c1 = centers[c] + offsets[c]
+        res[i * 3, 0] = a1
+        res[i * 3, 1] = (b1 - a1)
+        res[i * 3 + 1, 0] = b1
+        res[i * 3 + 1, 1] = (c1 - b1)
+        res[i * 3 + 2, 0] = c1
+        res[i * 3 + 2, 1] = (a1 - c1)
+    return res
```

Comment:
Perhaps replace `res` with `borders` for better reader comprehension::

```suggestion
    borders = np.empty((len(triangles)*3, 2, 2), dtype=centers.dtype)
    for i, triangle in enumerate(triangles):
        a, b, c = triangle
        a1 = centers[a] + offsets[a]
        b1 = centers[b] + offsets[b]
        c1 = centers[c] + offsets[c]
        borders[i * 3, 0] = a1
        borders[i * 3, 1] = (b1 - a1)
        borders[i * 3 + 1, 0] = b1
        borders[i * 3 + 1, 1] = (c1 - b1)
        borders[i * 3 + 2, 0] = c1
        borders[i * 3 + 2, 1] = (a1 - c1)
    return borders
```